File: src/main/java/io/vertigo/shiny/components/dataviz/chart/ShinyChart.java
package io.vertigo.shiny.components.dataviz.chart;

import java.util.List;

import io.vertigo.core.lang.Assertion;
import io.vertigo.shiny.components.ShinyComponent;

public record ShinyChart(
		String title,
		ShinyChartType chartType,
		List<String> labels,
		List<ShinyChartSerie> series) implements ShinyComponent {

	@Override
	public String type() {
		return switch (chartType) {
			case bar -> "ShinyBarChart";
			case area -> "ShinyAreaChart";
			case line -> "ShinyLineChart";
			case radar -> "ShinyRadarChart";
			case donut -> "ShinyDonutChart";
			case pie -> "ShinyPieChart";
		};
	}

	public ShinyChart {
		Assertion.check()
				.isNotBlank(title)
				.isNotNull(chartType)
				.isNotNull(labels)
				.isNotNull(series);
	}
}

File: banshee/src/models/ShinyChart.ts
import { ShinyComponent } from '../ShinyComponent';
import { ShinyChartType } from './ShinyChartType';
import { ShinyChartSerie } from './ShinyChartSerie';

export class ShinyChart implements ShinyComponent {
  title: string;
  chartType: ShinyChartType;
  labels: string[];
  series: ShinyChartSerie[];
  type: string;

  constructor(
    title: string,
    chartType: ShinyChartType,
    labels: string[],
    series: ShinyChartSerie[]
  ) {
    this.title = title;
    this.chartType = chartType;
    this.labels = labels;
    this.series = series;
    switch (chartType) {
      case ShinyChartType.BAR:
        this.type = 'barChart';
        break;
      case ShinyChartType.AREA:
        this.type = 'areaChart';
        break;
      case ShinyChartType.LINE:
        this.type = 'lineChart';
        break;
      case ShinyChartType.RADAR:
        this.type = 'radarChart';
        break;
      case ShinyChartType.DONUT:
        this.type = 'donutChart';
        break;
      case ShinyChartType.PIE:
        this.type = 'pieChart';
        break;
      default:
        this.type = 'chart';
    }
  }
}
